src/extlibs/hrvo/simulator.cpp

```diff
@@ -278,60 +282,48 @@ std::size_t HRVOSimulator::addLinearVelocityRobotAgent(const Robot &robot,
     // Enemy agents should appear larger to friendly agents to avoid collision
     float agent_radius = ROBOT_MAX_RADIUS_METERS * ENEMY_ROBOT_RADIUS_SCALE;
 
+    AgentPath path = addPath(destination, goal_radius);
     return addLinearVelocityAgent(position, agent_radius, velocity, max_speed, max_accel,
-                                  addGoal(destination), goal_radius);
+                                  path);
 }
 
 std::size_t HRVOSimulator::addHRVOAgent(const Vector &position, float agent_radius,
                                         const Vector &curr_velocity, float maxSpeed,
-                                        float prefSpeed, float maxAccel,
-                                        std::size_t goal_index, float goalRadius,
+                                        float prefSpeed, float maxAccel, AgentPath &path,
                                         float neighborDist, std::size_t maxNeighbors,
                                         float uncertaintyOffset)
 {
     std::shared_ptr<HRVOAgent> agent = std::make_shared<HRVOAgent>(
-        this, position, goal_index, neighborDist, maxNeighbors, agent_radius,
-        curr_velocity, maxAccel, goalRadius, prefSpeed, maxSpeed, uncertaintyOffset);
+        this, position, neighborDist, maxNeighbors, agent_radius, curr_velocity, maxAccel,
+        path, prefSpeed, maxSpeed, uncertaintyOffset);
     agents.push_back(std::move(agent));
     return agents.size() - 1;
 }
 
 size_t HRVOSimulator::addLinearVelocityAgent(const Vector &position, float agent_radius,
                                              const Vector &curr_velocity, float max_speed,
-                                             float max_accel, size_t goal_index,
-                                             float goal_radius)
+                                             float max_accel, AgentPath &path)
 {
     std::shared_ptr<LinearVelocityAgent> agent = std::make_shared<LinearVelocityAgent>(
-        this, position, agent_radius, curr_velocity, max_speed, max_accel, goal_index,
-        goal_radius);
+        this, position, agent_radius, curr_velocity, max_speed, max_accel, path);
 
     agents.push_back(std::move(agent));
     return agents.size() - 1;
 }
 
-std::size_t HRVOSimulator::addGoal(const Vector &position)
+AgentPath HRVOSimulator::addPath(const Vector &position, float goal_radius)
```

Comment:
We should remove the usage of this path with something like: `Path path = Path({PathPoint(position)}, goal_radius)`, (this is assuming that the speed at path point is 0).

Reply:
done